Mathlib/Algebra/Polynomial/Splits.lean

```diff
@@ -427,6 +427,19 @@ theorem mem_lift_of_splits_of_roots_mem_range [Algebra R K] {f : K[X]}
   obtain ⟨b, hb, rfl⟩ := Multiset.mem_map.1 hP
   exact Subring.sub_mem _ (X_mem_lifts _) (C'_mem_lifts (hr _ hb))
 
+/--
+A polynomial of degree `2` with a root splits.
+-/
+theorem splits_of_degree_two {f : Polynomial K} {x : L} (h₁ : f.natDegree = 2)
```

Comment:
```suggestion
theorem splits_of_natDegree_two {f : Polynomial K} {x : L} (h₁ : f.natDegree = 2)
```
(or `natDegree_eq_two`)
I wonder if a `f.natDegree <= 2` version would be easier to apply? Or is it false because of stupid reasons when ` = 0`?

Reply:
Constant polynomial are indeed splits, so I could indeed change to `f.natDegree <= 2`. Let me try that.

Reply:
After thinking about it, the assumption that the polynomial does have a root doesn't make sense for degree 0 and is not really relevant in degree 1, so I think it is better to keep it this way. I'll add a `degree` version of match what is done with [Polynomial.splits_of_natDegree_le_one](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Polynomial/Splits.html#Polynomial.splits_of_natDegree_le_one)